app/services/users.js

```diff
@@ -0,0 +1,25 @@
+const models = require('../models/index');
+const errors = require('../errors');
+const signUpValidator = require('../interactors/sign_up_validator');
+const generatePassword = require('../interactors/generate_password');
+
+module.exports.signUp = body =>
+  models.user.findOne({ where: { email: body.email } }).then(user => {
+    if (user) {
+      throw errors.emailInUseError();
+    } else {
+      signUpValidator.validate(body);
```

Comment:
Why validate this now? if the user makes a mistake it has to wait until you go look for the user in the database to get a response. Input validations should often be the first thing you check for. Also, since this validation does not need DB access, it should not be in the service, but the controller.